packages/mermaid/src/diagrams/xychart/xychartDb.ts

```diff
@@ -0,0 +1,302 @@
+
+const defaultThemeVariables = getThemeVariables();
+
+const config = configApi.getConfig();
+
+function plotColorPaletteGenerator(baseColor: string, noOfColorNeeded = 15): string[] {
+  const colors = [];
+  const MAX_HUE_VALUE = 360;
+  const baseHue = channel(baseColor, 'h');
+  if (baseHue > MAX_HUE_VALUE / 2) {
+    const decr = Math.floor(baseHue / noOfColorNeeded);
+    for (let i = 0; i <= baseHue; i += decr) {
+      colors.push(adjust(baseColor, { h: -i }));
+    }
+  } else {
+    const incr = Math.floor((MAX_HUE_VALUE - baseHue) / noOfColorNeeded);
+    for (let i = 0; i <= baseHue; i += incr) {
+      colors.push(adjust(baseColor, { h: i }));
+    }
+  }
+  return colors;
+}
+
+function getChartDefaultThemeConfig(): XYChartThemeConfig {
+  return {
+    xychartTitleColor:
+      config.themeVariables?.xychartTitleColor || defaultThemeVariables.xychartTitleColor,
+    xychartAxisLineColor:
+      config.themeVariables?.xychartAxisLineColor || defaultThemeVariables.xychartAxisLineColor,
+    xychartXAxisLableColor:
+      config.themeVariables?.xychartXAxisLableColor || defaultThemeVariables.xychartXAxisLableColor,
+    xychartXAxisTitleColor:
+      config.themeVariables?.xychartXAxisTitleColor || defaultThemeVariables.xychartXAxisTitleColor,
+    xychartXAxisTickColor:
+      config.themeVariables?.xychartXAxisTickColor || defaultThemeVariables.xychartXAxisTickColor,
+    xychartYAxisLableColor:
+      config.themeVariables?.xychartYAxisLableColor || defaultThemeVariables.xychartYAxisLableColor,
+    xychartYAxisTitleColor:
+      config.themeVariables?.xychartYAxisTitleColor || defaultThemeVariables.xychartYAxisTitleColor,
+    xychartYAxisTickColor:
+      config.themeVariables?.xychartYAxisTickColor || defaultThemeVariables.xychartYAxisTickColor,
+    xychartPlotBaseColor:
+      config.themeVariables?.xychartPlotBaseColor || defaultThemeVariables.xychartPlotBaseColor,
+  };
+}
+function getChartDefaultConfig(): XYChartConfig {
+  return config.xyChart
+    ? { ...config.xyChart, yAxis: { ...config.xyChart.yAxis }, xAxis: { ...config.xyChart.xAxis } }
+    : {
+        width: 700,
+        height: 500,
+        fontFamily: config.fontFamily || 'Sans',
+        titleFontSize: 16,
+        titlePadding: 5,
+        showtitle: true,
+        plotBorderWidth: 2,
+        yAxis: {
+          showLabel: true,
+          labelFontSize: 14,
+          lablePadding: 5,
+          showTitle: true,
+          titleFontSize: 16,
+          titlePadding: 5,
+          showTick: true,
+          tickLength: 5,
+          tickWidth: 2,
+        },
+        xAxis: {
+          showLabel: true,
+          labelFontSize: 14,
+          lablePadding: 5,
+          showTitle: true,
+          titleFontSize: 16,
+          titlePadding: 5,
+          showTick: true,
+          tickLength: 5,
+          tickWidth: 2,
+        },
+        chartOrientation: 'vertical',
+        plotReservedSpacePercent: 50,
+      };
```

Comment:
These values should be in the jsonSchema file and imported from the generated file.

Reply:
Done with the changes

Reply:
Fixed